components/deployment.sh

```diff
@@ -26,6 +26,11 @@ function itunes_connect() {
     --scheme "$scheme" \
     $clean_build
 
+  if [[ $build_number ]]; then
+    msg "Tagging build as $build_number"
+    git tag -a build-$build_number -m "Added by ios-tools"
```

Comment:
This is going to create LOTS of tags, depending on how it's used, but that's not clear. Should the circle.yml example be updated? I suggest this argument should only be used when deploying to iTC.

Also, I prefer string interpolation to be done inside double quotes.